src/main/groovy/io/cify/runner/utils/CapabilityParser.groovy

```diff
@@ -49,17 +62,37 @@ class CapabilityParser {
     }
 
     /**
-     * Gets all variations from capabilities file content
+     * Gets {@link Capabilities} base on {@link String} strategy specified
+     *
+     * @param capabilitiesFileContent capabilities file content
+     * @param strategy parsing strategy
+     * @return the list of {@link Capabilities}
+     */
+    private static List getCapabilities(LazyMap capabilitiesFileContent, String strategy) {
+        switch (strategy) {
+            case STRATEGY_VARIATIONS:
+                return getCapabilitiesVariations(capabilitiesFileContent)
+            case STRATEGY_ONE_BY_ONE:
+                return getCapabilitiesOneByOne(capabilitiesFileContent)
+            case STRATEGY_ALL_IN_ONE:
+                return getCapabilitiesAllInOne(capabilitiesFileContent)
+            default:
+                return getCapabilitiesVariations(capabilitiesFileContent)
+        }
+    }
+
+    /**
+     * Gets all variations of capabilities from capabilities file content
      *
      * @param capabilitiesFileContent - capabilities file content
-     * @return list of capabilities variations
-     * */
-    static List getCapabilitiesVariations(LazyMap capabilitiesFileContent) {
+     * @return the list of {@link Capabilities}
+     */
+    private static List getCapabilitiesVariations(LazyMap capabilitiesFileContent) {
```

Comment:
Maybe I forgot smth, but I capability per device or 1 capability per platform?

Reply:
I didn't get the question. Could you add more details?